### PR TITLE
ribbons.c; list_menu.c; cleanup previous PRs; knock out (most of) the rest of scrcmd.s

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/scrcmd_party.c [hunk 1]
```diff
@@ -659,7 +659,7 @@ BOOL ScrCmd_MonGetRibbonCount(SCRIPTCONTEXT *ctx) {
 
     u16 i, count;
     for (i = 0, count = 0; i < RIBBON_MAX; i++) {
-        if (GetMonData(pokemon, sub_0208E55C(i, 0), NULL)) {
+        if (GetMonData(pokemon, GetRibbonAttr(i, 0), NULL)) {
```
Comment (tgsm) [inline src/scrcmd_party.c:662]:
```suggestion
        if (GetMonData(pokemon, GetRibbonAttr(i, RIBBONDAT_MONDATNO), NULL)) {
```
(same goes for the other three instances in this file)

File: src/scrcmd_strbuf.c [hunk 2]
```diff
@@ -14,7 +14,7 @@ static STRING* _get_species_name(u16 species, HeapID heap_id);
 extern u16 GetStarterFromScriptState(SCRIPT_STATE*);
 extern u16 DPPtLeftover_GetRivalSpecies(SCRIPT_STATE*);
 extern u16 DPPtLeftover_GetFriendStarterSpecies(SCRIPT_STATE*);
-extern u32 sub_0208E55C(u8, u32);
+extern u32 GetRibbonAttr(u8, u32);
```
Comment (tgsm) [inline src/scrcmd_strbuf.c:17]: This can be removed

File: src/scrcmd_strbuf.c [hunk 3]
```diff
@@ -366,7 +366,7 @@ BOOL ScrCmd_BufferRibbonName(SCRIPTCONTEXT* ctx) {
     u8 idx = ScriptReadByte(ctx);
     u8 unk = ScriptGetVar(ctx);
 
-    u16 ribbon_id = sub_0208E55C(unk, 3);
+    u16 ribbon_id = GetRibbonAttr(unk, 3);
```
Comment (tgsm) [inline src/scrcmd_strbuf.c:369]:
```suggestion
    u16 ribbon_id = GetRibbonAttr(unk, RIBBONDAT_NAMEGMM);
```

File: src/trainer_data.c [hunk 4]
```diff
@@ -331,7 +331,7 @@ void CreateNPCTrainerParty(BATTLE_SETUP *enemies, int party_id, HeapID heap_id)
 
             // If you were treating the trainer type as a bitfield, you'd put the
             // checks for held item and moves here. You'd also treat the trpoke
-            // data as a flat u16 array rather than an array of fixed-width structs.
+            // data as a flat u16 array rather than an array of fixed-maxItemWidth structs.
```
Comment (tgsm) [inline src/trainer_data.c:334]: revert

File: src/unk_02055418.c [hunk 5]
```diff
@@ -76,7 +76,7 @@ void sub_02055508(FieldSystem* fsys, int unkA) {
 }
 
 void sub_020555B4(FieldSystem* fsys, u32 unkA, RTCTime* time) {
-    sub_02092E34(sub_02092DEC(), unkA, 0);
+    sub_02092E34(Fsys_GetGearPhoneRingManager(), unkA, 0);
```
Comment (tgsm) [inline src/unk_02055418.c:79]:
```suggestion
    sub_02092E34(Fsys_GetGearPhoneRingManager(fsys), unkA, 0);
```

File: src/unk_02092BE8.c [hunk 6]
```diff
@@ -279,47 +280,47 @@ BOOL sub_02093070(FieldSystem* sys){
...
     }
-    u16 var = (u16) (owned / 0x32);
-    if (!var) {
+    u16 var = (owned / 50);
+    if (var == 0) {
         return FALSE;
     }
     if (var > 9) {
-        var = 9;
+        var = 9; // uncreachable
```
Comment (tgsm) [inline src/unk_02092BE8.c:298]:
```suggestion
        var = 9; // unreachable
```

File: src/scrcmd_c.c [hunk 7]
```diff
@@ -843,3 +862,1020 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+    *p_work = NULL;
+    return TRUE;
+}
+
+BOOL ScrNative_WaitApplication(SCRIPTCONTEXT *ctx) {
+    if (!FieldSys_ApplicationIsRunning(ctx->fsys)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
```
Comment (tgsm) [inline src/scrcmd_c.c:1711]:
```suggestion
    return !FieldSys_ApplicationIsRunning(ctx->fsys);
```
`return FieldSys_ApplicationIsRunning(ctx->fsys) == FALSE;` also works

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/scrcmd_c.c (hunk 7 above)
Comment (PikalaxALT) [reply src/scrcmd_c.c:1721]: doesn't match tho

File: src/scrcmd_c.c [hunk 8]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+    GameStats_Add(Sav2_GameStats_get(ctx->fsys->savedata), statIdx, value);
+    return FALSE;
+}
+
+BOOL ScrCmd_SafariZoneAction(SCRIPTCONTEXT *ctx) {
+    FLYPOINTS_SAVE *flypointsSave = Save_FlyPoints_get(ctx->fsys->savedata); //r6
+    SCRIPT_STATE *scriptState = SavArray_Flags_get(ctx->fsys->savedata); //sp4
+    SAFARIZONE *safariZone = Save_SafariZone_get(ctx->fsys->savedata); //sp8
+    u8 action = ScriptReadByte(ctx); //r4
+    u8 areaSet = ScriptReadByte(ctx); //sp0
```
Comment (tgsm) [inline src/scrcmd_c.c:3389]: I'm assuming you don't need these register comments anymore

File: src/scrcmd_c.c [hunk 9]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+BOOL ScrCmd_600(SCRIPTCONTEXT *ctx) {
+    if (ov01_02205D68(ctx->fsys)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
+}
+
+BOOL ScrCmd_FollowPokeFacePlayer(SCRIPTCONTEXT *ctx) {
+    BOOL doFace = TRUE; //sp8
```
Comment (tgsm) [inline src/scrcmd_c.c:4471]: ditto

File: src/scrcmd_c.c [hunk 10]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+    return FALSE;
+}
+
+u32 GetMaxBankTransactionAmount(FieldSystem *fsys, int action) {
+    u32 ret;
+    u32 wallet = PlayerProfile_GetMoney(Sav2_PlayerData_GetProfileAddr(fsys->savedata));
+    u32 bank = MomSavingsBalanceAction(SaveData_GetMomsSavingsAddr(fsys->savedata), MOMS_BALANCE_GET, 0);
+    switch (action) {
+    case 0:
+        ret = 999999 - bank;
```
Comment (tgsm) [inline src/scrcmd_c.c:5066]:
```suggestion
        ret = MAX_MONEY - bank;
```

File: src/scrcmd_c.c [hunk 11]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+        if (wallet <= ret) {
+            ret = wallet;
+        }
+        break;
+    case 1:
+        ret = 999999 - wallet;
```
Comment (tgsm) [inline src/scrcmd_c.c:5072]: ditto

File: src/scrcmd_c.c [hunk 12]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+    FreeToHeap(work->sub);
+    FreeToHeap(*p_work);
+    return TRUE;
+}
+
+BOOL ScrCmd_BankOrWalletIsFull(SCRIPTCONTEXT *ctx) {
+    u16 action = ScriptReadHalfword(ctx);
+    u16 *p_ret = ScriptGetVarPointer(ctx);
+    if (action == 0) {
+        if (MomSavingsBalanceAction(SaveData_GetMomsSavingsAddr(ctx->fsys->savedata), MOMS_BALANCE_GET, 0) == 999999) {
```
Comment (tgsm) [inline src/scrcmd_c.c:5134]: ditto

File: src/scrcmd_c.c [hunk 13]
```diff
@@ -843,3 +927,4520 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
...
+            *p_ret = TRUE;
+        } else {
+            *p_ret = FALSE;
+        }
+    } else {
+        if (PlayerProfile_GetMoney(Sav2_PlayerData_GetProfileAddr(ctx->fsys->savedata)) == 999999) {
```
Comment (tgsm) [inline src/scrcmd_c.c:5140]: ditto

**Review by tgsm — CHANGES_REQUESTED** [review]
My remaining two comments from the last review should be addressed, too

File: src/trainer_data.c (hunk 4 above)
Comment (ScottNortonPhD) [reply src/trainer_data.c:334]: Have I mentioned that CLion's struct field renaming is overzealous?

**Review by tgsm — APPROVED** [review]

**Pull request merged**